src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/ScoreboardEvents.kt

```diff
@@ -0,0 +1,500 @@
+}
+
+private fun getNewYearLines(): List<String> {
+    return listOf(getSbLines().first { SbPattern.newYearPattern.matches(it) })
+}
+
+private fun getNewYearShowWhen(): Boolean {
+    return getSbLines().any { SbPattern.newYearPattern.matches(it) }
+}
+
+private fun getOringoLines(): List<String> {
+    return listOf(getSbLines().first { SbPattern.travelingZooPattern.matches(it) })
+}
+
+private fun getOringoShowWhen(): Boolean {
+    return getSbLines().any { SbPattern.travelingZooPattern.matches(it) }
+}
+
+private fun getMiningEventsLines() = buildList {
+    // Wind
+    if (getSbLines().any { SbPattern.windCompassPattern.matches(it) }
+        && getSbLines().any { SbPattern.windCompassArrowPattern.matches(it) }) {
+        add(getSbLines().first { SbPattern.windCompassPattern.matches(it) })
+        add("| ${getSbLines().first { SbPattern.windCompassArrowPattern.matches(it) }} §f|")
+    }
+
+    // Better Together
+    if (getSbLines().any { SbPattern.nearbyPlayersPattern.matches(it) }) {
+        add("§dBetter Together")
+        add(" ${getSbLines().first { SbPattern.nearbyPlayersPattern.matches(it) }}")
+    }
+
+    // Zone Events
+    if (getSbLines().any { SbPattern.miningEventPattern.matches(it) }
+        && getSbLines().any { SbPattern.miningEventZonePattern.matches(it) }) {
+        add(getSbLines().first { SbPattern.miningEventPattern.matches(it) }.removePrefix("Event: "))
+        add("in ${getSbLines().first { SbPattern.miningEventZonePattern.matches(it) }.removePrefix("Zone: ")}")
+    }
+
+    // Zone Events but no Zone Line
+    if (getSbLines().any { SbPattern.miningEventPattern.matches(it) }
+        && getSbLines().none { SbPattern.miningEventZonePattern.matches(it) }) {
+        add(getSbLines().first { SbPattern.miningEventPattern.matches(it) }
+            .removePrefix("Event: "))
+    }
+
+    // Mithril Gourmand
+    if (getSbLines().any { SbPattern.mithrilRemainingPattern.matches(it) }
+        && getSbLines().any { SbPattern.mithrilYourMithrilPattern.matches(it) }) {
+        add(getSbLines().first { SbPattern.mithrilRemainingPattern.matches(it) })
+        add(getSbLines().first { SbPattern.mithrilYourMithrilPattern.matches(it) })
+    }
+
+    // Raffle
+    if (getSbLines().any { SbPattern.raffleTicketsPattern.matches(it) }
+        && getSbLines().any { SbPattern.rafflePoolPattern.matches(it) }) {
+        add(getSbLines().first { SbPattern.raffleTicketsPattern.matches(it) })
+        add(getSbLines().first { SbPattern.rafflePoolPattern.matches(it) })
+    }
+
+    // Raid
+    if (getSbLines().any { SbPattern.yourGoblinKillsPattern.matches(it) }
+        && getSbLines().any { SbPattern.remainingGoblinPattern.matches(it) }) {
+        add(getSbLines().first { SbPattern.yourGoblinKillsPattern.matches(it) })
+        add(getSbLines().first { SbPattern.remainingGoblinPattern.matches(it) })
+    }
+}
+
+private fun getMiningEventsShowWhen(): Boolean {
+    return inMiningIsland()
```

Comment:
there are no events in deep cavern or golden mine. Oh wait the function isn't merged yet in beta🤦‍♂️
Just add an todo
```suggestion
    return inAdvancedMiningIsland()
```